### PR TITLE
[SYCL-MLIR] Add parallel_for.cpp

### PR DESCRIPTION
**Pull request by whitneywhtsang**
Signed-off-by: Tsang, Whitney <whitney.tsang@intel.com>

### CODE REVIEW

File: SYCL/Basic/cgeist/parallel_for.cpp [hunk 1]
```diff
@@ -0,0 +1,36 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple -Xcgeist -gen-all-sycl-funcs %s -o %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+// RUN: %ACC_RUN_PLACEHOLDER %t.out
+// REQUIRES: linux
+// UNSUPPORTED: hip || cuda
+
+#include <sycl/sycl.hpp>
+using namespace sycl;
+#define N 8
```
Comment (victor-eds) [inline SYCL/Basic/cgeist/parallel_for.cpp]: Can we have this defined as `static constexpr` instead?

**Review by victor-eds — APPROVED** [review]
LGTM but nit

File: SYCL/Basic/cgeist/parallel_for.cpp (hunk 1 above)
Comment (whitneywhtsang) [reply SYCL/Basic/cgeist/parallel_for.cpp]: Sure. https://stackoverflow.com/questions/1637332/static-const-vs-define explains their pros and cons.

**Comment by etiotto** [comment]
For the record IMO we should not add test cases that fail because it makes PRs appear as "red". The correct process is to add test coverage only after the PR that makes the test case pass has been merged in.

**Pull request merged**